### PR TITLE
[Storage-Queue][Storage-File] Kwargify positional params

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 1]
```diff
@@ -81,11 +81,11 @@ def __init__( # type: ignore
             directory_path=None, # type: Optional[str]
             snapshot=None,  # type: Optional[Union[str, Dict[str, Any]]]
             credential=None, # type: Optional[Any]
-            loop=None,  # type: Any
             **kwargs # type: Optional[Any]
         ):
         # type: (...) -> None
         kwargs['retry_policy'] = kwargs.get('retry_policy') or ExponentialRetry(**kwargs)
+        loop = kwargs.pop('loop', None)
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Add :keyword loop: to docstring?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 2]
```diff
@@ -506,7 +503,7 @@ def list_handles(self, recursive=False, timeout=None, **kwargs):
             file.
         :param int max_concurrency:
             Maximum number of parallel connections to use.
-        :param int timeout:
+        :keyword int timeout:
             The timeout parameter is expressed in seconds.
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Need to change metadata, content_settings, validate_content, max_concurrency, and encoding to :keyword: as well.

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 3]
```diff
@@ -547,7 +550,7 @@ def list_handles(self, recursive=False, timeout=None, **kwargs):
 
         :param str file_name:
             The name of the file to delete.
-        :param int timeout:
+        :keyword int timeout:
             The timeout parameter is expressed in seconds.
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Looks like we're keeping this timeout as a param.

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py [hunk 4]
```diff
@@ -130,11 +130,11 @@ def __init__(  # type: ignore
         file_path=None,  # type: Optional[str]
         snapshot=None,  # type: Optional[Union[str, Dict[str, Any]]]
         credential=None,  # type: Optional[Any]
-        loop=None,  # type: Any
         **kwargs  # type: Any
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py]: keyword loop in docstring?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py [hunk 5]
```diff
@@ -409,6 +408,7 @@ def __init__(  # type: ignore
         :type copy_id: str or ~azure.storage.file.FileProperties
         :rtype: None
         """
+        timeout = kwargs.pop('timeout', None)
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py]: :keyword int timeout: not in docstring

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py [hunk 6]
```diff
@@ -77,11 +77,11 @@ class FileServiceClient(AsyncStorageAccountHostsMixin, FileServiceClientBase):
     def __init__(
             self, account_url,  # type: str
             credential=None,  # type: Optional[Any]
-            loop=None,  # type: Any
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py]: loop docstring

File: sdk/storage/azure-storage-file/azure/storage/file/aio/share_client_async.py [hunk 7]
```diff
@@ -516,22 +521,24 @@ def list_directories_and_files( # type: ignore
             process_storage_error(error)
 
     @distributed_trace_async
-    async def create_directory(self, directory_name, metadata=None, timeout=None, **kwargs):
+    async def create_directory(self, directory_name, **kwargs):
         # type: (str, Optional[Dict[str, Any]], Optional[int], Any) -> DirectoryClient
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/share_client_async.py]: we can remove Optional[int] in type hint

File: sdk/storage/azure-storage-file/azure/storage/file/file_client.py [hunk 8]
```diff
@@ -933,10 +940,10 @@ def upload_range_from_url(self, source_url,  # type: str
             The range can be up to 4 MB in size.
             The start_range and end_range params are inclusive.
             Ex: start_range=0, end_range=511 will download first 512 bytes of file.
-        :param int timeout:
+        :keyword int timeout:
             The timeout parameter is expressed in seconds.
         '''
-
+        timeout = kwargs.pop('timeout', None)
```
Comment (kristapratico) [inline sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: timeout isn't being passed to any method

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py (hunk 3 above)
Comment (kristapratico) [reply sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: On second thought, was this one just missed being put into kwargs?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py [hunk 9]
```diff
@@ -244,16 +244,18 @@ def list_shares(
                 :dedent: 8
                 :caption: Create a share in the file service.
         """
+        metadata = kwargs.pop('metadata', None)
+        quota = kwargs.pop('quota', None)
+        timeout = kwargs.pop('timeout', None)
         share = self.get_share_client(share_name)
         kwargs.setdefault('merge_span', True)
-        await share.create_share(metadata, quota, timeout, **kwargs)
+        await share.create_share(metadata=metadata, quota=quota, timeout=timeout, **kwargs)
```
Comment (xiangyan99) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py]: Why not just **kwargs?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py [hunk 10]
```diff
@@ -109,6 +111,7 @@ def __init__(
                 :dedent: 8
                 :caption: Get file service properties.
         """
+        timeout = kwargs.pop('timeout', None)
         try:
             return await self._client.service.get_properties(timeout=timeout, **kwargs)
```
Comment (xiangyan99) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_service_client_async.py]: Why not just **kwargs?

File: sdk/storage/azure-storage-file/azure/storage/file/directory_client.py [hunk 11]
```diff
@@ -543,22 +544,23 @@ def create_subdirectory(
                 :dedent: 12
                 :caption: Create a subdirectory.
         """
+        metadata = kwargs.pop('metadata', None)
+        timeout = kwargs.pop('timeout', None)
         subdir = self.get_subdirectory_client(directory_name)
         subdir.create_directory(metadata=metadata, timeout=timeout, **kwargs)
```
Comment (xiangyan99) [inline sdk/storage/azure-storage-file/azure/storage/file/directory_client.py]: Same

Comment (rakshith91) [reply sdk/storage/azure-storage-file/azure/storage/file/directory_client.py]: Updated everywhere, and history.md

**Comment by rakshith91** [comment]
CI is green
https://dev.azure.com/azure-sdk/public/_build/results?buildId=146882&view=results

**Pull request merged**